### PR TITLE
add inspircd docs

### PR DESCRIPTION
**Pull request by butlerx**

### CODE REVIEW

File: docs/services/irc.md [hunk 1]
```diff
@@ -11,19 +14,20 @@ Most of the stuff in ircd.conf is fairly obvious. For adding new external users
...
+  immediately be kicked off the network. This gets old pretty quick.
 
 #### Compiling
 
 ```bash
 $ cd /tmp; mkdir build; cd build
 
 # may want to delete the other files that you get with it for clarity
-$ apt-get source ircd-hybrid 					
+$ apt-get source ircd-hybrid
```
Comment (VoyTechnology) [inline docs/services/irc.md]: Can we just use `apt` instead?

File: docs/services/irc.md [hunk 2]
```diff
@@ -59,3 +64,165 @@ chan_is_opless(const struct Channel *const chptr)
...
+    build: '.'
+    container_name: 'irc'
+    restart: 'always'
+    ports:
+      - 6697:6697
+      - 7000:7000
+      - 7005:7005
+    volumes:
+      - '/etc/inspircd:/inspircd/conf:ro'
+      - '/etc/apache2/ssl:/ssl
```
Comment (VoyTechnology) [inline docs/services/irc.md:126]: Is it safe sharing a container running as root with our SSL cert? Probably better than it is now, but still something to consider

File: docs/services/irc.md (hunk 1 above)
Comment (butlerx) [reply docs/services/irc.md]: no  these are the old docs ircd only runs on older insatnces

**Comment by VoyTechnology** [comment]
Resolves #23

File: docs/services/irc.md (hunk 2 above)
Comment (butlerx) [reply docs/services/irc.md:126]: possibly not the container can be easily run as another user. if we do we should add a script to make sure the file permissions of the configs are preserved. i beleive @d-fens has a nice script for this

**Review by gruunday — APPROVED** [review]

**Comment by butlerx** [comment]
im merging this in since this is the current setup and the rest of the setup will be handled by @gruunday @CSmartt @AlexKraak

**Pull request merged**